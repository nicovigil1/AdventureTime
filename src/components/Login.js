import React from 'react'

export default class Login extends React.Component {
  getLogin = () => {
    let getLoginText = document.querySelector("#textBox").value
    let url = `https://adventure-time-m4cap.herokuapp.com/api/v1/login?username=${getLoginText}`;
    fetch(url, { method: "POST" })
      .then(res => res.json())
      .then(person => this.props.getUser(person.Value))
      .then(this.props.display)
      .then(this.props.getLocation) // .then(postLocation(this.props.getLocation))
  }
  postLocation = (geoResponse) => {
    //fetch(`https://adventure-time-m4cap.herokuapp.com/locationEndpoint` {method: "POST", body: geoResponse})
  }
  render() {
    return(
      <div style={styleLogin}>
        <p>Sign in Below!</p>
        <input id="textBox" type="text"/>
        <button onClick={this.getLogin} style={styleSubmit}>Login</button>
      </div>
    )
  }
}


const styleLogin = {
  display: "flex", 
  flexDirection: "column",
  textAlign: "center",
  padding: "2em",
  background: "rgba(206, 215, 219, 0.96)",
  borderRadius: "1em"
}

const styleSubmit = {
  marginTop: ".4em",
  marginLeft: "2em",
  marginRight: "2em"
}