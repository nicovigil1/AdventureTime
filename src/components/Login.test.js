import React from 'react'
import Login from './Login'
import {shallow} from 'enzyme'
import { isMainThread } from 'worker_threads';


describe('Header', () => {
  let wrapper 
  let display = jest.fn()
  let getUser = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<Login display={display} getUser={getUser}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})