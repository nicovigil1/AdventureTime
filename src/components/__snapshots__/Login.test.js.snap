// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<div
  style={
    Object {
      "background": "rgba(206, 215, 219, 0.96)",
      "borderRadius": "1em",
      "display": "flex",
      "flexDirection": "column",
      "padding": "2em",
      "textAlign": "center",
    }
  }
>
  <p>
    Sign in Below!
  </p>
  <input
    id="textBox"
    type="text"
  />
  <button
    onClick={[Function]}
    style={
      Object {
        "marginLeft": "2em",
        "marginRight": "2em",
        "marginTop": ".4em",
      }
    }
  >
    Login
  </button>
</div>
`;
